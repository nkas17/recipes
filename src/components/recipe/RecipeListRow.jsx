import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

/**
 * A Row of a recipe
 */
const RecipeListRow = ({ recipe, deleteRecipe, showDelete }) => (
  <tr>
    <td>
      <Link to={`recipe/${recipe.id}`}>{recipe.title}</Link>
    </td>
    <td className="recipe-description">{recipe.description}</td>
    <td>{recipe.category}</td>
    {showDelete && (
      <td>
        <input
          type="button"
          value="delete"
          className="btn btn-link delete-recipe"
          onClick={() => deleteRecipe(recipe._id)}
        />
      </td>
    )}
  </tr>
);

RecipeListRow.propTypes = {
  /**
   * the recipe to display
   */
  recipe: PropTypes.objectOf(PropTypes.any).isRequired,

  /**
   * function to delete a recipe
   */
  deleteRecipe: PropTypes.func.isRequired,

  /**
   * whether or not we should show the delete button
   */
  showDelete: PropTypes.bool.isRequired,
};

export default RecipeListRow;
