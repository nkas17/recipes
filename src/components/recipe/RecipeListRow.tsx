import React from 'react';
import { Link } from 'react-router-dom';
import { Recipe } from '../../types/recipe';

interface RecipeListRowProps {
  recipe: Recipe;
  deleteRecipe: (id: { $oid: string } | undefined) => void;
  showDelete: boolean;
}

/**
 * A Row of a recipe
 */
function RecipeListRow({ recipe, deleteRecipe, showDelete }: RecipeListRowProps) {
  return (
    <tr>
      <td>
        <Link to={`recipe/${recipe.id}`}>{recipe.title}</Link>
      </td>
      <td className="recipe-description">{recipe.description}</td>
      <td>{recipe.category}</td>
      {showDelete && (
        <td>
          <input
            type="button"
            value="delete"
            className="btn btn-link delete-recipe"
            onClick={() => deleteRecipe(recipe._id)}
          />
        </td>
      )}
    </tr>
  );
}

export default RecipeListRow;
