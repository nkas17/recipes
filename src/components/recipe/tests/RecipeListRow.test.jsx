import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import RecipeListRow from '../RecipeListRow';

const recipe = {
	id: 'tacos',
	title: 'tacos',
	category: 'other',
	description: 'Tasty homemade tacos - hard or soft',
	ingredients: ['cheese', 'ground beef', 'tortilla'],
};

describe('RecipeListRow', () => {
	it('renders correctly', () => {
		const tree = shallow(
			<RecipeListRow recipe={recipe} showDelete deleteRecipe={() => {}} />
		);
		expect(toJSON(tree)).toMatchSnapshot();
	});
});
