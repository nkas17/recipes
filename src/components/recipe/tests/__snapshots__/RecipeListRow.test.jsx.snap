// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeListRow renders correctly 1`] = `
<tr>
  <td>
    <Link
      to="recipe/tacos"
    >
      tacos
    </Link>
  </td>
  <td
    className="recipe-description"
  >
    Tasty homemade tacos - hard or soft
  </td>
  <td>
    other
  </td>
  <td>
    <input
      className="btn btn-link delete-recipe"
      onClick={[Function]}
      type="button"
      value="delete"
    />
  </td>
</tr>
`;
